--- a/arch/arm64/boot/dts/qcom/sm6115-fxtec-pro1x.dts
+++ b/arch/arm64/boot/dts/qcom/sm6115-fxtec-pro1x.dts
@@ -11,6 +11,7 @@
 #include <dt-bindings/arm/qcom,ids.h>
 #include <dt-bindings/leds/common.h>
 #include <dt-bindings/usb/pd.h>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
 
 / {
 	model = "F(x)tec Pro1X (QX1050)";
@@ -34,6 +35,30 @@
 			clocks = <&gcc GCC_DISP_HF_AXI_CLK>;
 		};
 	};
+	/* Panel Power Supply: 5V rail for the display */
+	panel_edp_supply: panel-edp-supply {
+		compatible = "regulator-fixed";
+		regulator-name = "panel_edp_supply";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		gpio = <&tlmm 100 GPIO_ACTIVE_HIGH>;
+		startup-delay-us = <20000>;
+	};
+
+	dsi_panel_pinctrl: dsi-panel-pinctrl {
+		compatible = "qcom,pm6150l-dsi-pinctrl";
+
+		dsi_panel_reset: dsi-panel-reset-pins {
+			pins = "gpio1";
+			function = "gpio";
+
+			config {
+				qcom,drive-strength = <12>;
+				qcom,pull = <0>;
+				qcom,output-high;
+			};
+		};
+	};
 
 	disp_elvdd_supply: disp-elvdd-supply {
 		compatible = "regulator-fixed";
@@ -168,6 +193,9 @@
 &mdss_dsi0 {
 	vdda-supply = <&pm6125_l18a>;
 	status = "okay";
+	qcom,num-dsi-ports = <1>;
+	qcom,dsi-lane-map = <0 1 2 3>;
+	qcom,num-dsi-lanes = <4>;
 
 	panel: panel@0 {
 		compatible = "boe,bf060y8m-aj0";
@@ -175,6 +203,8 @@
 
 		reset-gpios = <&tlmm 82 GPIO_ACTIVE_LOW>;
 
+		vdds-supply = <&panel_edp_supply>; /* 5V panel supply */
+
 		elvdd-supply = <&disp_elvdd_supply>;
 		elvss-supply = <&disp_elvss_supply>;
 		vcc-supply = <&disp_vcc_supply>;
@@ -184,6 +214,8 @@
 		pinctrl-0 = <&mdss_dsi_n &panel_en_n>;
 		pinctrl-names = "default";
 
+		status = "okay";
+
 		port {
 			panel_in: endpoint {
 				remote-endpoint = <&mdss_dsi0_out>;
